contracts/SEURO.sol

```diff
@@ -4,44 +4,53 @@ pragma solidity ^0.8.2;
 import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import "@openzeppelin/contracts/access/AccessControl.sol";
 
-// import "hardhat/console.sol";
-
 contract SEuro is ERC20, AccessControl {
   bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
   bytes32 public constant BURNER_ROLE = keccak256("BURNER_ROLE");
-  // bytes32 public constant DEFAULT_ADMIN_ROLE = keccak256("DEFAULT_ADMIN_ROLE");
+  bytes32 public constant DAO_ADMIN = keccak256("DAO_ADMIN");
 
   constructor(string memory name, string memory symbol, address[] memory _admins)
-      public
       ERC20(name, symbol)
   {
     _grantRole(MINTER_ROLE, msg.sender);
     _grantRole(BURNER_ROLE, msg.sender);
-
-    _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    _grantRole(DAO_ADMIN, msg.sender);
 
     for(uint8 i=0; i<_admins.length; i++){
       _grantRole(MINTER_ROLE, _admins[i]);
       _grantRole(BURNER_ROLE, _admins[i]);
     }
   }
 
+  modifier onlyAdmin() {
+      require(hasRole(DAO_ADMIN, msg.sender), "invalid-admin");
+      _;
+  }
+
   function mint(address to, uint256 amount) public {
     require(hasRole(MINTER_ROLE, msg.sender), "invalid-minter");
     _mint(to, amount);
   }
 
+  function addMinter(address _address) public onlyAdmin {
+      _grantRole(MINTER_ROLE, _address);
```

Comment:
i think we can leverage AccessControl better if we use `grantRole` here, instead of `_grantRole`
that function is already only accessible for admins for that role, and there would be no need for our own `onlyAdmin` modifier

Reply:
fair point 👍 all for better safety